lib/web/users.go

```diff
@@ -169,7 +169,13 @@ func updateUser(r *http.Request, m userAPIGetter) (*ui.User, error) {
 
 	user.SetRoles(req.Roles)
 
-	updateUserTraits(req, user)
+	// we do not want situation where one trait set (allTraits)
+	// overrides another (traits).
+	if len(req.AllTraits) > 0 {
+		user.SetTraits(req.AllTraits)
+	} else {
+		updateUserTraits(req, user)
+	}
```

Comment:
Is this API covered by unit tests? Can we cover this logic as well?

Reply:
tests added in https://github.com/gravitational/teleport/pull/42620/commits/0d6b589bcda5f4c5525ffba6964a281b30d7025b#diff-182f90cb2b8f651ffc752d871baae2b6de277eec5a40fbb4cbbc28c8648b4676